pkg/ebpf/dns_cache_test.go

```diff
@@ -169,6 +169,32 @@ func TestDNSCacheTelemetry(t *testing.T) {
 	assert.Equal(t, expected, cache.Stats())
 }
 
+func TestDNSCacheMerge(t *testing.T) {
+	ttl := 100 * time.Millisecond
+	cache := newReverseDNSCache(1000, ttl, disableAutomaticExpiration)
+
+	ts := time.Now()
+	conns := []ConnectionStats{
+		{
+			Source: util.AddressFromString("127.0.0.1"),
+			Dest:   util.AddressFromString("192.168.0.1"),
+		},
+	}
+
+	t1 := newTranslation([]byte("host-b"))
+	t1.add(util.AddressFromString("192.168.0.1"))
+	cache.Add(t1, ts)
+	res := cache.Get(conns, ts)
+	assert.Equal(t, []string{"host-b"}, res[util.AddressFromString("192.168.0.1")])
+
+	t2 := newTranslation([]byte("host-a"))
+	t2.add(util.AddressFromString("192.168.0.1"))
+	cache.Add(t2, ts)
+	res = cache.Get(conns, ts)
```

Comment:
Could we also add a `t3` re-adding the same translation to make sure the result won't contain duplicate entries?

Reply:
Done